### ISSUE TITLE
Bulk inserts for relationship

### ISSUE THREAD

**Issue by yos1p** [laravel/ideas#2596]
As you know, there will be quite latency for Laravel (basically any other framework too) to execute multiple queries.
Therefore, this:
```
INSERT INTO user_adress (user_id, address_id) VALUES (1, 2), (1, 3), (1,4)
```
is better than this:
```
INSERT INTO user_address (user_id, address_id) VALUES (1,2)
INSERT INTO user_address (user_id, address_id) VALUES (1,3)
INSERT INTO user_address (user_id, address_id) VALUES (1,4)
```

I notice that syncWithoutDetaching insert the data one by one, instead of in one query. I haven't checked sync, attach, or detach yet. What do you guys think?

I can open PR if needed.

**Comment by paras-malhotra** [comment]
Laravel already supports bulk inserts. Just use `Model::insert`.

**Comment by yos1p** [comment]
I will rephrase it to make it clearer for anyone else.

The idea is not to bulk inserts for Model, that is already covered with `Model::insert`. I know that. But what about the relationship?

Usually, you would do something like:
`$user->address()->syncWithoutDetaching($addressIds)` 

In Laravel, these executed as multiple queries, instead of just one. To deal with this, you can do something like:
`DB::table('user_address')->insert($relationships)`

So, wouldn't it be better if the relationships also inserted in bulk? Instead of one by one?

**Comment by themsaid** [comment]
Feel free to open a PR with your proposed changes.
